### PR TITLE
getDecade and getLustrum based formatting settings for year notches

### PR DESCRIPTION
**Pull request by palewire**

### CODE REVIEW

File: public/javascripts/timeline-setter.js [hunk 1]
```diff
@@ -240,13 +246,40 @@
...
       var idx = this.idx;
       var date = new Date(ts);
       while(idx--){
         var intvl = this.INTERVAL_ORDER[idx];
-        date["set" + intvl](intvl === "Date" ? 1 : 0);
+        switch(intvl){
+          case 'Lustrum':
+            date["setFullYear"](this.getDecade(date));
+            break;
+          case 'FullYear':
```
Comment (thejefflarson) [inline public/javascripts/timeline-setter.js:277]: Shouldn't this be `case 'Lustrum':` ? and the one above `case 'Decade':`

**Comment by thejefflarson** [comment]
Hmm. Are the switch statements off by one?

File: public/javascripts/timeline-setter.js [hunk 2]
```diff
@@ -240,13 +246,40 @@
...
+        return (date.getFullYear() / 10 | 0) * 10;
+    },
+    
+    // Returns the first year of the five year "lustrum" a Date belongs to
+    // as an integer. A lustrum a fancy Roman word for a "five-year period."
+    // You can read more about it here: http://en.wikipedia.org/wiki/Lustrum
+    // This all means that if you pass in the year 2011 you'll get 2010 back.
+    // And if you pass in the year 1997 you'll get 1995 back.
+    getLustrum : function(date) {
+        return ((date.getFullYear() / 5 | 0) * 10) / 2;
```
Comment (thejefflarson) [inline public/javascripts/timeline-setter.js:264]: Shouldn't this be `(date.getFullYear() / 5 | 0) * 5` ?

Comment (palewire) [inline public/javascripts/timeline-setter.js:264]: As long as the math works and rounds down to the nearest interval of five, that's fine with me.

File: public/javascripts/timeline-setter.js (hunk 1 above)
Comment (thejefflarson) [inline public/javascripts/timeline-setter.js:277]: oh wait I see what the issue is. yeah this is a problem here. let me see if I can find a solution.

Comment (palewire) [inline public/javascripts/timeline-setter.js:277]: I think the loops run differently for some reason. Frankly, I didn't figure it all out -- just kludged till I thought I got it to work right.

File: public/javascripts/timeline-setter.js (hunk 2 above)
Comment (thejefflarson) [inline public/javascripts/timeline-setter.js:264]: heh, 10 /  2 === 5

File: public/javascripts/timeline-setter.js (hunk 1 above)
Comment (thejefflarson) [inline public/javascripts/timeline-setter.js:277]: yeah we talked about it and I think it works correctly. The only thing we have to be careful about is not setting FullYear to 0.

File: public/javascripts/timeline-setter.js (hunk 2 above)
Comment (palewire) [inline public/javascripts/timeline-setter.js:264]: If any method deserves a little needless mystery, it's getLustrum.

**Pull request merged**